stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -452,23 +488,48 @@ Base.@constprop :aggressive function generic_matmatmul_wrapper!(C::StridedMatrix
         return _rmul_or_fill!(C, β)
     end
     matmul2x2or3x3_nonzeroalpha!(C, tA, tB, A, B, α, β) && return C
-    # We convert the chars to uppercase to potentially unwrap a WrapperChar,
-    # and extract the char corresponding to the wrapper type
-    tA_uc, tB_uc = uppercase(tA), uppercase(tB)
     alpha, beta = promote(α, β, zero(T))
-    if alpha isa Union{Bool,T} && beta isa Union{Bool,T}
-        if tA_uc == 'S' && tB_uc == 'N'
-            return BLAS.symm!('L', tA == 'S' ? 'U' : 'L', alpha, A, B, beta, C)
-        elseif tA_uc == 'N' && tB_uc == 'S'
-            return BLAS.symm!('R', tB == 'S' ? 'U' : 'L', alpha, B, A, beta, C)
-        elseif tA_uc == 'H' && tB_uc == 'N'
-            return BLAS.hemm!('L', tA == 'H' ? 'U' : 'L', alpha, A, B, beta, C)
-        elseif tA_uc == 'N' && tB_uc == 'H'
-            return BLAS.hemm!('R', tB == 'H' ? 'U' : 'L', alpha, B, A, beta, C)
-        end
+    blasfn = _valtypeparam(val)
+    if alpha isa Union{Bool,T} && beta isa Union{Bool,T} && blasfn ∈ (BlasFlag.SYMM, BlasFlag.HEMM)
+        _blasfn = blasfn
+        αβ = (alpha, beta)
+    else
+        _blasfn = BlasFlag.NONE
+        αβ = (α, β)
+    end
+    _symm_hemm_generic!(C, tA, tB, A, B, αβ..., Val(_blasfn))
+    return C
+end
+function _lrchar_ulchar(tA, tB)
+    if uppercase(tA) == 'N'
+        lrchar = 'R'
+        ulchar = isuppercase(tB) ? 'U' : 'L'
+    else
+        lrchar = 'L'
+        ulchar = isuppercase(tA) ? 'U' : 'L'
     end
-    return _generic_matmatmul!(C, wrap(A, tA), wrap(B, tB), MulAddMul(α, β))
+    return lrchar, ulchar
 end
+function _symm_hemm_generic!(C, tA, tB, A, B, alpha, beta, ::Val{BlasFlag.SYMM})
+    lrchar, ulchar = _lrchar_ulchar(tA, tB)
+    if lrchar == 'L'
+        BLAS.symm!(lrchar, ulchar, alpha, A, B, beta, C)
+    else
+        BLAS.symm!(lrchar, ulchar, alpha, B, A, beta, C)
+    end
+end
+function _symm_hemm_generic!(C, tA, tB, A, B, alpha, beta, ::Val{BlasFlag.HEMM})
+    lrchar, ulchar = _lrchar_ulchar(tA, tB)
+    if lrchar == 'L'
+        BLAS.hemm!(lrchar, ulchar, alpha, A, B, beta, C)
+    else
+        BLAS.hemm!(lrchar, ulchar, alpha, B, A, beta, C)
+    end
+end
+Base.@constprop :aggressive function _symm_hemm_generic!(C, tA, tB, A, B, α, β, ::Val{BlasFlag.NONE})
+    _generic_matmatmul!(C, wrap(A, tA), wrap(B, tB), MulAddMul(α, β))
```

Comment:
Do we want/need the "`@stable_muladdmul`" (or whatever it's called) macro here?

Reply:
Yes, you're right, although I think it might make sense to narrow its scope similar to https://github.com/JuliaLang/julia/pull/56289